src/integrations/prefect-aws/prefect_aws/experimental/bundles/execute.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import annotations
+
+import os
+import tempfile
+from pathlib import Path
+
+import typer
+from botocore.exceptions import ClientError
+from pydantic_core import from_json
+
+from prefect.runner import Runner
+from prefect.utilities.asyncutils import run_coro_as_sync
+from prefect_aws.credentials import AwsCredentials
+
+from .types import AwsCredentialsBlockName, S3Bucket, S3Key
+
+
+def download_bundle_from_s3(
+    bucket: S3Bucket,
+    key: S3Key,
+    output_dir: str | None = None,
+    aws_credentials_block_name: AwsCredentialsBlockName | None = None,
+) -> dict[str, str]:
+    """
+    Downloads a bundle from an S3 bucket.
+
+    Args:
+        bucket: S3 bucket name
+        key: S3 object key
+        output_dir: Local directory to save the bundle (if None, uses a temp directory)
+        aws_credentials_block_name: Name of the AWS credentials block to use
+    """
+
+    if aws_credentials_block_name:
+        aws_credentials = AwsCredentials.load(aws_credentials_block_name)
+    else:
+        aws_credentials = AwsCredentials()
+
+    s3 = aws_credentials.get_s3_client()
+
+    output_dir = output_dir or tempfile.mkdtemp(prefix="prefect-bundle-")
+    Path(output_dir).mkdir(parents=True, exist_ok=True)
+
+    local_path = Path(output_dir) / os.path.basename(key)
+
+    try:
+        s3.download_file(bucket, key, local_path)
+        return {"local_path": local_path}
+    except ClientError as e:
+        raise RuntimeError(f"Failed to download bundle from S3: {e}")
+
+
+def execute_bundle_from_s3(
+    bucket: S3Bucket,
+    key: S3Key,
+    aws_credentials_block_name: AwsCredentialsBlockName | None = None,
+) -> None:
+    """
+    Downloads a bundle from S3 and executes it.
+
+    This step:
+    1. Downloads the bundle from S3
+    2. Extracts and deserializes the bundle
+    3. Executes the flow in a subprocess
+
+    Args:
+        bucket: S3 bucket name
+        key: S3 object key
+        aws_credentials_block_name: Name of the AWS credentials block to use
+    """
+    download_result = download_bundle_from_s3(
+        bucket=bucket,
+        key=key,
+        aws_credentials_block_name=aws_credentials_block_name,
+    )
+
+    bundle_data = from_json(Path(download_result["local_path"]).read_bytes())
```

Comment:
given the reliance on the `local_path` key, could be worth elevating the return type of `download_bundle` to a `TypedDict` or `dataclass`